examples/6.6 - BinarySearch/BinarySearch.sc

```diff
@@ -3,7 +3,7 @@ def binarySearch[T: Ordering](sorted: IndexedSeq[T], target: T): Boolean = {
     // If the sequence you are trying to search has no items, the item cannot be found
     if (start == end) false
     else {
-      val middle = (start + end) / 2
+      val middle = start + (end - start) / 2
```

Comment:
A more concise way used in JDK: `val middle = (start + end) >>> 1`

JDK16 source code:
https://github.com/openjdk/jdk16/blob/4de3a6be9e60b9676f2199cd18eadb54a9d6e3fe/src/java.base/share/classes/java/util/Collections.java#L228